### ISSUE TITLE
Apply Suggestions? Cannot see ability to Apply all

### ISSUE THREAD

**Issue by LeoJHarris** [sharpenrocks/Sharpen#22]
Hi there, have started using this with Visual Studio 2017, regarding the apply solution found here on your wiki page:

https://raw.githubusercontent.com/ironcev/sharpen/master/images/code-refactoring-on-arbitrary-scale.png

When I mouse over to try to get the ability to "apply suggestions" I cant get that option dropdown? How can I apply all rather then individually?

**Comment by ironcev** [comment]
Hi @LeoJHarris,

Great to hear that you've started using Sharpen! Unfortunately, the refactoring feature is still not available :-( I've mentioned that on the README page:

> Note that in the current version of Sharpen the following features mentioned in the above chapters are still not available:
> 
> Extensive documentation.
> Recommendations.
> Refactoring.
> Configuration.

And also on the [Refactoring Code](https://github.com/sharpenrocks/Sharpen/wiki/Refactoring-Code) Wiki page:

> At the moment Sharpen does not have any refactorings implemented. Refactorings are planned and will be implemented in the upcoming versions.
> Meanwhile, for some of the suggestions, you can use built-in Visual Studio refactorings.

The "screenshot" is actually a photoshopped image supposed to give an idea of what should be a part of Sharpen.
But I think I have to make this note/disclaimer more prominent. On the README page I'll add a visible title "Current Limitations".

FYI next week I am back to Europe and plan to continue with the work on Sharpen. Current roadmap is VS 2019 support, embedded documentation, and after that refactoring. Please stay patient.

Thanks,
Igor

**Comment by LeoJHarris** [comment]
@ironcev my apologisies i did not see that, really appreciate your effort and look forward to this and will wait on. Thanks

**Comment by LeoJHarris** [comment]
Hi @ironcev might this still be implemented at some stage? Am staying patient but this would be a super cool feature that I'm still hoping to see! 🥇

**Comment by ironcev** [comment]
Hi @LeoJHarris, thanks for pinging me and motivating me at the same time!

To reflect on my own words:

> FYI next week I am back to Europe and plan to continue with the work on Sharpen. Current roadmap is VS 2019 support, embedded documentation, and after that refactoring. Please stay patient.

VS2019 support was done mid of February as planned. But afterward, I, unfortunately, had a long pause in the development of almost two and a half months :-( I got out of "hibernation" this week actually. The roadmap slightly changed in between. Support for C# 8.0 is my top priority at the moment. On Saturday I am giving a conference talk on C# 8.0 and want to use Sharpen during the talk. As a teaser :-) here is what already works in alpha quality:

![image](https://user-images.githubusercontent.com/4142833/57491667-f8cbc500-72bd-11e9-8ec8-12d52d4b83c4.png)

(Suggestions for async streams are there as well but do not appear in the real-life result shown on this screenshot.)

I expect to polish the C# 8.0 suggestions earliest by the end of May.

The good news is, the demand for refactoring came from several sides. The general feedback I got is that refactoring would significantly increase Sharpen's usefulness so I plan to shift them before the embedded documentation.

Long story short: yes it will be implemented at some stage, hopefully starting this summer.

Thanks for being patient so far and once again for providing feedback! Also, if you have a hint which refactorings you would like to see first, please let me know.

**Comment by sturlath** [comment]
Hi hi any news on this? 

I would love to be able to work through the list in the Sharpen Results view and just select "Apply suggestions"

**Comment by ironcev** [comment]
Hi @sturlath, no, unfortunately, no news so far :-( The last Sharpen release was in October last year and I struggle ever since to continue with the development. The last few months were packed with activities that let me very little time for Sharpen. The good news is, I am catching up with open source contributions and when it comes to Sharpen the feedback I got from several sources tells the same - "Apply Suggestions" is the feature users mostly ask for. So I plan to tackle this feature first once I am back to the development.

And thank you for asking! These kinds of "pinging" motivate me to guard time for Sharpen and continue with the development. Thanks!

**Comment by sturlath** [comment]
Yes please keep on working on Sharpen! Its a awesome tool and with the apply functionality it will probably skyrocket in use!

I'll ping you until its finished no worries :-)

**Comment by LeoJHarris** [comment]
Hi @ironcev any movement yet or potential opportunity for this to be included shortly? 🔥
